### DISCUSSION TITLE
TFMini LiDAR modul support

### DISCUSSION THREAD

**Discussion by speedymk1** [arendst/Tasmota#17384] [Ideas]
Hello together,

I love the Tasmota project and use it for my SonOff devices.
Is it possible that someone implement an TFMini LiDAR modul?
I found [this](https://github.com/arendst/Tasmota/discussions/10543) discussion, but I am not a good programmer and so I would like to ask you if it could be a "standard" I2C sensor.

I like to measure how wide open my garage door is (50 cm or 100 cm or totally open?).
It will be nice, if someone could implement it. I will do the testing so far I am able.

here you are able to find a datasheet and description here:
[datasheet and description](https://www.reichelt.at/at/de/benewake-tfmini-s-lidar-12m-tf-mini-s-p287742.html?PROVID=2807&gclid=Cj0KCQiA4uCcBhDdARIsAH5jyUlQKRh6z3HsmkTCtbyH-3nw6i2YZWdvgTBoALZvNE9yli4unG6-4_kaAj9oEALw_wcB)

**Comment by sfromis** [comment]
Sure it is possible that "someone implement", if that someone has the device, interest and skills to sit down and code a driver. If you use ESP32, that's not as bad as it sounds, as the Berry language is then available, being reasonably friendly for coding an I2C driver, without diving into the rabbit hole of getting used to C++.

**Comment by barbudor** [comment]
Tasmota is supporting the TF-Mini+ Lidar sensor since March 2021
It is not included in any standard built so you may need to compile it yourself

However it is included in the unofficial All-Sensors variant that you can flash from the Tasmota Web Installer

**Reply by barbudor** [reply]
Note that other ToF sensords are also available for the same usage such as the popular VL053X

**Reply by speedymk1** [reply]
Thanks, I will try it tonight

**Reply by speedymk1** [reply]
Hi again,
I am sorry, but the webinstaller don't work for me, I tried it serveral times.
But tasmotizer-1.2.exe work.

My problem is now, that I don't know where I could download the unofficial All-Sensors binary.
Please help me again.

**Reply by sfromis** [reply]
The Tasmota Web Installer loads that "unofficial" image from here:
https://github.com/tasmota/install/tree/main/firmware/unofficial

**Reply by speedymk1** [reply]
Thanks for all your assistance, but Tasmota didn't find the TF-Mini+ on the I2C bus.
I have to find another solution.

**Reply by sfromis** [reply]
You should not expect it on the I2C bus, as indicated in #15305 you need to config tx/rx pins, as the [driver](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xsns_sensor/xsns_86_tfminiplus.ino) is using serial.

And the pins are included in the allsensors build dropdowns for gpio on Configure Module / Template.

However, near the end of the linked thread, you can also find an I2C driver for the device, coded in Berry, available if/when you use ESP32. I do not know how well-tested that sample code is....

**Reply by speedymk1** [reply]
Hi sfromis,

thanks, I will give it a last try.
I thought it is I2C, however
I use an ESP 12E node mcu board.
Now I flashed following:

```
Program Version:       12.3.1.1(allsensors)
Build Date & Time:     2022-12-16T09:51:10
Core/SDK Version:      2_7_4_9/2.2.2-dev(38a443e)
...
ESP Chip Id:          5017702 (ESP8266EX)
Flash Chip Id:        0x164020 (DOUT)
Flash Size:           4096 KB
Program Flash Size:   4096 KB
Program Size:         737 KB
Free Program Size:    264 KB
Free Memory:          16.7 KB
```

I have done the same like descripted in [#15305](https://github.com/arendst/Tasmota/discussions/15305) but without success.
I give up and will find another solution...

But I will say thanks to all of you.

**Reply by barbudor** [reply]
If you want I2C distance sensors, look at VL53L0X and VL53L1X supported by Tasmota

**Reply by sfromis** [reply]
It happens that people claiming "I have done the same like", without describing what they did, actually did not follow what the other one was doing, meaning can get lost in words, and "without success" tells nothing about where you stumbled. Minute details matter a lot
